### PR TITLE
Create automated tests

### LINKED ISSUE #27: Create automated tests
we need some QA smh jp is asking for it

### PR DESCRIPTION
**Pull request by Henriquelay**
Setups Jest to work properly with Typescript.

TODO: Testing setup using database and endpoints that modify the database

### CODE REVIEW

**Review by AtilioA — DISMISSED** [review]

**Comment by AtilioA** [comment]
LGTM for now

**Review by Henriquelay — COMMENTED** [review]
Meu pau fi, difícil pra crl automar teste com DB kkkk o resto tá tudo pronto ja e tô aqui nessa luta

**Comment by AtilioA** [comment]
pera vc tava falando daqui
foca em fazer os unitários funcionarem

**Comment by Henriquelay** [comment]
> pera vc tava falando daqui foca em fazer os unitários funcionarem

Os unitários estão bala e exemplos estão já pushados o foda tá sendo testar com DB

File: server/src/app.test.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
-import CreateServer from './app';
+import app from './app';
 
 it('Installs all the middleware', () => {
-  const app = CreateServer();
   expect(app.listen).toBeDefined();
-  expect(app.use).toHaveLength(4);
+  // I don't even know what else to test
```
Comment (AtilioA) [inline server/src/app.test.ts:5]: I think it is fine as it is. There's not much to do here.

File: server/src/database/index.ts [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+    });
+  },
+
+  async drop(connectionName?: string) {
+    await getConnection(connectionName).dropDatabase();
+  },
+};
+
+DB.connect().catch(err => {
+  console.error('❌ Error connection to DB!');
```
Comment (AtilioA) [inline server/src/database/index.ts]: connecting :rose:

File: server/src/middlewares/ensureAuthenticated.test.ts [hunk 3]
```diff
@@ -2,6 +2,21 @@ import { Request, Response } from 'express';
 import ensureAuthenticated from './ensureAuthenticated';
 
 describe('ensures', () => {
+  // FIXME not actually throwing WTF but when env is actually no set on server it throws. I don't think it have anything to do with asynchronous tests and such, since they are on different threads.
+  it.skip('fails when `JWT_SECRET` env is not set', () => {
+    const request = {
+      // Is there a way to modularize this better? Doesn't feel DRY enough.
```
Comment (AtilioA) [inline server/src/middlewares/ensureAuthenticated.test.ts:8]: modularize _WHAT_?

File: server/src/middlewares/ensureAuthenticated.test.ts [hunk 4]
```diff
@@ -2,6 +2,21 @@ import { Request, Response } from 'express';
 import ensureAuthenticated from './ensureAuthenticated';
 
 describe('ensures', () => {
+  // FIXME not actually throwing WTF but when env is actually no set on server it throws. I don't think it have anything to do with asynchronous tests and such, since they are on different threads.
```
Comment (AtilioA) [inline server/src/middlewares/ensureAuthenticated.test.ts:5]: I guess it won't throw unless the JWT token is missing. Invalid tokens, etc are caught by our try-catch; only missing tokens throw errors (`Error`).
Is `authConfig.jwt.secret` set and loaded correctly? I don't think it is related but as a last resort...

File: server/src/middlewares/ensureAuthenticated.test.ts [hunk 5]
```diff
@@ -15,6 +30,7 @@ describe('ensures', () => {
       next,
     );
     expect(next.mock.calls).toHaveLength(1);
+    // REVIEW - I don't know how to hard type `jest.fn.mock.calls` since they are `any[]`
```
Comment (AtilioA) [inline server/src/middlewares/ensureAuthenticated.test.ts:33]: :rose: why does my 4k display snap from time to time

File: server/src/middlewares/ensureAuthenticated.ts [hunk 6]
```diff
@@ -39,7 +39,7 @@ export default function ensureAuthenticated(
     }
     const [, token] = authHeader.split(' '); // Do not use "type" of token (Bearer)
 
-    const decoded = verify(token, authConfig.jwt.secret as string);
+    const decoded = verify(token, authConfig.jwt.secret);
```
Comment (AtilioA) [inline server/src/middlewares/ensureAuthenticated.ts:42]: This was throwing errors before. Is it working as-is now?

File: server/src/middlewares/requestsLogger.ts [hunk 7]
```diff
@@ -14,10 +14,14 @@ export default function logRequest(
   _: Response,
   next: NextFunction,
 ): void {
-  const { method, url } = request;
-  const logLabel = `[${method.toUpperCase()}] request to ${url}`;
+  if (process.env.NODE_ENV !== 'test') {
```
Comment (AtilioA) [inline server/src/middlewares/requestsLogger.ts:17]: :100:

File: server/src/middlewares/ensureAuthenticated.test.ts (hunk 3 above)
Comment (Henriquelay) [reply server/src/middlewares/ensureAuthenticated.test.ts:8]: The middleware parameters. Request, Reponse and next. I think I did something of the sorts on another project, but I don't have access to the source anymore

File: server/src/middlewares/ensureAuthenticated.test.ts (hunk 4 above)
Comment (Henriquelay) [reply server/src/middlewares/ensureAuthenticated.test.ts:5]: Not the JWT token, the JWT_SECRET env

File: server/src/middlewares/ensureAuthenticated.test.ts (hunk 5 above)
Comment (Henriquelay) [reply server/src/middlewares/ensureAuthenticated.test.ts:33]: 👏

File: server/src/middlewares/ensureAuthenticated.ts (hunk 6 above)
Comment (Henriquelay) [reply server/src/middlewares/ensureAuthenticated.ts:42]: ESLint accused that authConfig was simply already a string

**Comment by Henriquelay** [comment]
Think it's already enough to start solving for #33 ?

**Comment by AtilioA** [comment]
I don't remember where this is standing and I haven't _tested_ it myself, but if it is working for unit tests then I'd say yes

**Comment by Henriquelay** [comment]
Check status in issue main message. Improvements on DB testing could be made.

**Comment by AtilioA** [comment]
Fine by me, we can improve it later

**Comment by AtilioA** [comment]
MIGATIONS :fire:

**Comment by AtilioA** [comment]
Make India Great Again

**Review by AtilioA — APPROVED** [review]

**Pull request merged**